engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreEngineActor.scala

```diff
@@ -0,0 +1,166 @@
+package cromwell.engine.workflow.workflowstore
+
+import akka.actor.{ActorLogging, ActorRef, LoggingFSM, Props}
+import cats.data.NonEmptyList
+import cromwell.core.Dispatcher._
+import cromwell.engine.workflow.WorkflowManagerActor
+import cromwell.engine.workflow.WorkflowManagerActor.WorkflowNotFoundException
+import cromwell.engine.workflow.workflowstore.WorkflowStoreEngineActor.WorkflowStoreActorState
+import cromwell.engine.workflow.workflowstore.WorkflowStoreState.StartableState
+import WorkflowStoreEngineActor._
+import cromwell.core.WorkflowId
+import cromwell.engine.workflow.workflowstore.WorkflowStoreActor._
+import cromwell.services.metadata.MetadataService.MetadataPutAcknowledgement
+import org.apache.commons.lang3.exception.ExceptionUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+final case class WorkflowStoreEngineActor(store: WorkflowStore, serviceRegistryActor: ActorRef)
```

Comment:
This is mostly the same as the original `WorkflowStoreActor`, I forget the exact changes but it was mostly to make the wiring work.